File: app/src/main/java/com/jinhanexample/designPattern/adapter/java/operation/OperationAdaptee.java
package com.jinhanexample.designPattern.adapter.java.operation;

public class OperationAdaptee {
    public static final int ADD_OPERATION = 1;
    public static final int SUBTRACT_OPERATION = 2;
    public static final int MULTIPLY_OPERATION = 3;
    public static final int DIVIDE_OPERATION = 4;

    public OperationAdaptee() {
        super();
    }

    public int calculate(int operatorType, int firstNumber, int secondNumber) {
        switch (operatorType) {
            case ADD_OPERATION:
                return firstNumber + secondNumber;
            case SUBTRACT_OPERATION:
                return firstNumber - secondNumber;
            case MULTIPLY_OPERATION:
                return firstNumber * secondNumber;
            case DIVIDE_OPERATION:
                return firstNumber / secondNumber;
        }
        return 0;
    }
}

File: app/src/main/java/com/jinhanexample/designPattern/adapter/kotlin/operation/OperationAdaptee.kt
package com.jinhanexample.designPattern.adapter.kotlin.operation

class OperationAdaptee {

    companion object {
        const val Add_Operation = 1
        const val SUBTRACT_OPERATION = 2
        const val MULTIPLY_OPERATION = 3
        const val DIVIDE_OPERATION = 4
    }

    fun calculate(operationType: Int, firstNumber: Int, secondNumber: Int): Int {
        return when (operationType) {
            Add_Operation -> {
                firstNumber + secondNumber
            }
            SUBTRACT_OPERATION -> {
                firstNumber - secondNumber
            }
            MULTIPLY_OPERATION -> {
                firstNumber * secondNumber
            }
            DIVIDE_OPERATION -> {
                firstNumber / secondNumber
            }
            else -> 0
        }


    }
}